src/Microsoft.Data.Analysis/PrimitiveColumnContainerHelpers.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Data.Analysis
+{
+    internal static class PrimitiveColumnContainerHelpers
+    {
+        internal static DataFrameBuffer<T> GetMutable<T>(this IList<ReadOnlyDataFrameBuffer<T>> bufferList, int index)
```

Comment:
I would consider having the word "Copy" somewhere in the name, just so the caller is clear that they're going to copy a potentially large amount of data.

Reply:
Perhaps `GetOrCreateMutable`.